Common/OpenCL/ITKimprovements/itkOpenCL.h

```diff
@@ -20,6 +20,11 @@
 
 #include "itkOpenCLExport.h"
 
+// For the Official Khronos OpenCL headers
+#ifndef CL_TARGET_OPENCL_VERSION
+#  define CL_TARGET_OPENCL_VERSION 120
+#endif
+
```

Comment:
For the record, we had already some discussion about `CL_TARGET_OPENCL_VERSION` at pull request https://github.com/SuperElastix/elastix/pull/213 "Khronos OpenCL support", December 2019

Reply:
Yes, that PR has not been merged, but the issue still exists.

Reply:
Thanks Matt. So now I still wonder how your proposed `CL_TARGET_OPENCL_VERSION` relates to the `OPENCL_C_VERSION_1_1`, `CL_TARGET_OPENCL_VERSION_1_2`, and `OPENCL_C_VERSION_2_0` CMake variables from elastix. Should the definition of `CL_TARGET_OPENCL_VERSION` correspond to the enabled `OPENCL_C_VERSION` CMake variable(s)? Or are they entirely independent?

Is `120` just a default, or is it the only proper value for `CL_TARGET_OPENCL_VERSION` right now?

Reply:
`OPENCL_C_VERSION*` were designed for building against Intel, Nvidia, or AMD vendor toolchains:

https://github.com/SuperElastix/elastix/blob/be0abdcbd8a47837b62310a458c56c9f1579015b/CMake/FindOpenCL.cmake#L410

https://github.com/SuperElastix/elastix/blob/be0abdcbd8a47837b62310a458c56c9f1579015b/CMake/FindOpenCL.cmake#L473-L493

`CL_TARGET_OPENCL_VERSION` is intended for the Khronos OpenCL headers per:

https://github.com/KhronosGroup/OpenCL-Headers/blob/master/README.md#compiling-for-a-specific-opencl-version

`120` is a good default / proper value because it is the supported value across most implementations. On macOS in particular, support ended at this version.